metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1244,56 +1245,77 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode) -> SqlDataSet
         parent_data_set = node.parent_node.accept(self)
         parent_alias = self._next_unique_table_alias()
 
-        # Build time spine dataset
-        metric_time_dimension_instance: Optional[TimeDimensionInstance] = None
-        for instance in parent_data_set.metric_time_dimension_instances:
-            if len(instance.spec.entity_links) == 0:
-                # Use the instance with the lowest granularity
-                if not metric_time_dimension_instance or (
-                    instance.spec.time_granularity < metric_time_dimension_instance.spec.time_granularity
-                ):
-                    metric_time_dimension_instance = instance
         assert (
-            metric_time_dimension_instance
-        ), "Can't join to time spine without metric time. Validations should have prevented this."
-        metric_time_dimension_column_name = self.column_association_resolver.resolve_spec(
-            metric_time_dimension_instance.spec
+            len(node.requested_agg_time_dimension_specs) > 0
+        ), "Must have at least one value in requested_agg_time_dimension_specs for JoinToTimeSpineNode."
+
+        # Determine if the time spine join should use metric_time or the agg_time_dimension (metric_time takes priority).
+        agg_time_dimension_for_join = node.requested_agg_time_dimension_specs[0]
+        for spec in node.requested_agg_time_dimension_specs[1:]:
```

Comment:
This is a little magical - how do we know the first one is always not-metric-time and the others are always metric time? What if the user just requested metric_time, wouldn't the first one be potentially lowest-granularity metric_time?

I think it'll be cleaner to add a property to the node that gives us back the spec we need here, then we can be intentional with the naming and establish what people should care about. Like if all we need is the name/link pattern of an agg time dimension spec we can even return just those things.

Reply:
We don't know if the first one is always not-metric-time, I just overlooked checking the first one 🤦 

Re: granularity - the purpose of this check is just to see if the user passed `metric_time` or `agg_time_dimension` - it doesn't care about the granularity, so it stops as soon as it finds ANY `metric_time`. The granularity check happens a few lines down when going through all time dimension instances in the parent data set.

But I agree with your point about more intentional naming - will clean that up!

Reply:
Alright, I added a new property to the node called `query_contains_metric_time` and removed this logic!